CDCgov/prime-reportstream
# 14084 - Track interactions with the new Search and Filter features on the Daily Data page

This PR ...

Adds tracking on Submit, Apply, and Reset

Test Steps:
1. Log into RS
2. Go to /daily-data
3. Perform a search
4. Perform a filter, click Apply
5. Click Reset
6. Check App Insights for event logs

## Changes
![Screenshot 2024-05-23 at 3 18 40 PM](https://github.com/CDCgov/prime-reportstream/assets/102491809/bec50722-8c54-4ab1-a36d-ef4ab0c5d3bc)
![Screenshot 2024-05-23 at 3 19 32 PM](https://github.com/CDCgov/prime-reportstream/assets/102491809/c3f775ca-885a-45b0-9489-d03d2ad92a7a)

## Checklist

### Testing

## Linked Issues
- Fixes #14084
